packages/lodestar/src/db/api/beacon/repository.ts

```diff
@@ -65,17 +65,22 @@ export abstract class BulkRepository<T> extends Repository<T> {
   }
 
   public async getAllBetween(lowerLimit: number|null, upperLimit: number|null, step: number|null = null): Promise<T[]> {
+    const safeLowerLimit = lowerLimit || Buffer.alloc(0);
+    const safeUpperLimit = upperLimit || Number.MAX_SAFE_INTEGER;
     const data = await this.db.search({
-      gt: encodeKey(this.bucket, lowerLimit || Buffer.alloc(0)),
-      lt: encodeKey(this.bucket, upperLimit || Number.MAX_SAFE_INTEGER),
+      gt: encodeKey(this.bucket, safeLowerLimit),
+      lt: encodeKey(this.bucket, safeUpperLimit),
     });
-    let processedData = data;
-    if (step !== null) {
-      processedData = data.filter((datum, index) => {
-        return index % step === 0;
+    const processedData = (data || [])
+      .map((processedData) => deserialize(processedData, this.type))
+      .filter(block => {
```

Comment:
This method is not block specific, and this variabkle doesn't have slot field.

Reply:
I think we should override the `getAllBetween` method in src/db/api/beacon/repositories/blockArchive.ts w an implementation like this, where we know we're only handling blocks.